### PR TITLE
Move api host ports filtering on management space to the bootstrap worker

### PR DESCRIPTION
**Pull request by nvinuesa**
The function `FilterHostPortsForManagementSpace()`, which was added on a previous patch to the new network domain, is actually not needed as a service function. We refactor things in order to only use the spaces we already have during bootstrap in the bootstrap worker.

Note: this patch originates from the comment https://github.com/juju/juju/pull/16832#discussion_r1477339103

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```
juju bootstrap localhost c
juju add-model m
juju deploy ubuntu
```
```
go test github.com/juju/juju/internal/worker/bootstrap/... -gocheck.v  
```

## Links

**Jira card:** JUJU-5324

### CODE REVIEW

**Review by manadart — COMMENTED** [review]
Looks OK, but you need to regenerate some mocks.

**Comment by nvinuesa** [comment]
@manadart fixed the generated mocks, thanks!

File: internal/worker/bootstrap/worker.go [hunk 1]
```diff
@@ -250,14 +250,47 @@ func (w *bootstrapWorker) initAPIHostPorts(ctx context.Context, controllerConfig
...
+		if mgmtSpaceInfo == nil {
+			return apiHostPorts
+		}
+		hostPortsForAgents = make([]network.SpaceHostPorts, len(apiHostPorts))
+		for i := range apiHostPorts {
+			filtered, addrsIsInSpace := apiHostPorts[i].InSpaces(*mgmtSpaceInfo)
+			if addrsIsInSpace {
+				hostPortsForAgents[i] = filtered
+			} else {
+				hostPortsForAgents[i] = apiHostPorts[i]
```
Comment (manadart) [inline internal/worker/bootstrap/worker.go:286]: One last thing. Log a warning here.

**Review by manadart — APPROVED** [review]
Thanks for fixing this.

**Pull request merged**